calico_containers/tests/st/test_arg_parsing.py

```diff
@@ -0,0 +1,99 @@
+from sh import ErrorReturnCode_1, ErrorReturnCode_255
+
+from test_base import TestBase
+from docker_host import DockerHost
+
+
+class TestArgParsing(TestBase):
+    def test_arg_parsing(self):
+        """
+        Test that calicoctl correctly accepts or rejects given argument.
+        """
+        host = DockerHost('host')
+
+        calicoctl = "/code/dist/calicoctl %s"
+
+        # Run various commands with invalid IPs.
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "node --ip=127.a.0.1")
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "node --ip=aa:bb::cc")
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "node --ip=127.0.0.1 --ip6=127.0.0.1")
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "node --ip=127.0.0.1 --ip6=aa:bb::zz")
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "bgppeer rr add 127.a.0.1")
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "bgppeer rr add aa:bb::zz")
+        with self.assertRaises(ErrorReturnCode_255):
+            host.execute(calicoctl % "pool add 127.a.0.1")
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "pool add aa:bb::zz")
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "container node1 ip add 127.a.0.1")
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "container node1 ip add aa:bb::zz")
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "container add node1 127.a.0.1")
+        with self.assertRaises(ErrorReturnCode_1):
+            host.execute(calicoctl % "container add node1 aa:bb::zz")
+
+        # Add some pools and BGP peers and check the show commands"
+        host.execute(calicoctl % "bgppeer rr add 1.2.3.4")
+        host.execute(calicoctl % "bgppeer rr show")
```

Comment:
Probably originally so the developer could see the output. Removing them since they're asserted on.